precommit.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+"""
+Runs precommit checks on the repository.
+"""
+import argparse
+import concurrent.futures
+import hashlib
+import os
+import pathlib
+import subprocess
+import sys
+from typing import List, Union, Tuple  # pylint: disable=unused-import
+
+import yapf.yapflib.yapf_api
+
+
+def compute_hash(text: str) -> str:
```

Comment:
Please remove the hashing. Have a look at some other open-sourced projects (spurplus? icontract?) and adapt their precommit.py.